sympy/matrices/expressions/matexpr.py

```diff
@@ -372,6 +373,41 @@ def _eval_derivative(self, v):
         from sympy import KroneckerDelta
         return KroneckerDelta(self.args[1], v.args[1])*KroneckerDelta(self.args[2], v.args[2])
 
+    def _MatrixElement_expansion(self, printer, elementAccess):
+        '''
+        _MatrixElement_expansion distributes the element access over the terms in the MatrixElement object.
+        printer is the requesting printer to generate the expanded form of the MatrixElement object.
+        elementAccess is the precomputed indices to access the matrix element from the matrix.
+        '''
+        from sympy.matrices import MatrixSymbol, MutableDenseMatrix
+
+        # if the MatrixElement object is an instance of MatrixSymbol,
+        # simply, return the concatenation of the matrix element and elementAccess
+        if isinstance(self.parent, MatrixSymbol):
+             return printer._print(self.parent) + elementAccess
+
+        # get the terms from the MatrixElement object
+        terms = list(self.parent.args)
+        # creat an empty list in which the intermediate result is stored
+        l = []
+        # loop over the terms and concatenate the term and elementAccess
```

Comment:
This is way too complicated, this method should just rewrite `MatrixElement(MatAdd(a, b, c,  ...), ...)` to `Add(MatrixElement(a, ...), MatrixElement(b, ...), ... )`